core/contracts/financial_templates/FeePayer.sol

```diff
@@ -0,0 +1,79 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../FixedPoint.sol";
+import "../Finder.sol";
+import "../Testable.sol";
+import "../StoreInterface.sol";
+
+contract FeePayer is Testable {
+    using SafeMath for uint;
+    using FixedPoint for FixedPoint.Unsigned;
+
+    /**
+     * The collateral currency used to back the positions in this contract.
+     */
+
+    IERC20 public collateralCurrency;
+
+    /**
+     * Finder contract used to look up addresses for UMA system contracts.
+     */
+    Finder public finder;
+
+    /**
+     * Tracks the last block time when the fees were paid.
+     */
+    uint public lastPaymentTime;
+
+    /**
+     * @notice modifier that calls payFees().
+     */
+    modifier fees {
```

Comment:
maybe rename to more intuitive `paysFees` or something similar? (i.e. it denotes that certain methods will pay oracle fees)

Reply:
I was going to call it that, but that conflicts with the payFees method name :)

Reply:
Will try to think of a better name.

Reply:
feePayer? 
yeah low priority honestly just being nitpicky

Reply:
yeah, but feePayer is the name of the *contract* haha. I can't think of a good name that doesn't conflict with other names in the contract